src/natcap/invest/scenic_quality/scenic_quality.py

```diff
@@ -852,7 +855,7 @@ def _clip_and_mask_dem(dem_path, aoi_path, target_path, working_dir):
     dem_nodata = dem_raster_info['nodata'][0]
 
     def _mask_op(dem, aoi_mask):
-        valid_pixels = ((dem != dem_nodata) &
+        valid_pixels = (~utils.array_equals_nodata(dem, dem_nodata) &
                         (aoi_mask == 1))
```

Comment:
> Kinda makes me wish we had static typing

Yes. I felt myself thinking the same thing. It also made me think that we should be really good about using model defined constant nodata values well in the future. Like `TARGET_FLOAT_MIN32_NODATA` or something, so that for downstream operations it is obvious what we're comparing against.

In this case, I couldn't be sure if `dem` was float or int, or if `dem_nodata` was the input raster nodata or a nodata that we had defined in an intermediate step.

Reply:
> In this case, I couldn't be sure if `dem` was float or int, or if `dem_nodata` was the input raster nodata or a nodata that we had defined in an intermediate step.

Agreed. I just think it looks kinda funny followed by `aoi_mask == 1`. Clearly we're confident about the types of those variables, but not about the types of `dem` and `dem_nodata`. For me it kind of highlights how it could make development easier & less buggy to know all the types by looking at the code.